addons/web_editor/controllers/main.py

```diff
@@ -519,8 +520,8 @@ def shape(self, module, filename, **kwargs):
         """
         svg = None
         if module == 'illustration':
-            attachment = request.env['ir.attachment'].sudo().search([('url', '=like', request.httprequest.path), ('public', '=', True)], limit=1)
-            if not attachment:
+            attachment = request.env['ir.attachment'].sudo().browse(unslug(filename)[1])
```

Comment:
do we still need the sudo ? if public, not needed imo

Perso, since filename is a path and not a string, it can contains '/' so I prefer your previous version:
_, id = unslug(filename.split('/')[-1])

Reply:
sudo is still needed because ir_attachment explicitly defines the public portal users access rights to 0, 0, 0, 0

Reply:
@JKE-be if it contains a slash, would it not be an invalid slug?

Reply:
Yes, in standard case it will not happen.
I'm speaking about the guys that come with invalid link or manually crafted url.

Since we have already a test that return 404, better to use it that return a 403 or 500